app/components/SwitchSetting.js

```diff
@@ -0,0 +1,65 @@
+import { Platform, StyleSheet, View } from "react-native";
+import { Text } from "./index";
+import { Switch } from "@rneui/themed";
+import React, { useContext } from "react";
+import { ThemeContext } from "../theme-context";
+
+const SwitchSetting = ({ title, description, value, onValueChange }) => {
+  const { theme } = useContext(ThemeContext);
+  const s = getStyles(theme);
+
+  return (
+    <View>
+      <View style={[s.title, { marginBottom: 4 }]}>
+        <Text style={s.titleText}>{title}</Text>
+      </View>
+      <View
+        style={{
+          flexDirection: "row",
+          alignItems: "center",
+          marginVertical: -8,
+        }}
+      >
+        <Text style={[s.descriptionText, { flex: 1 }]}>{description}</Text>
+        <Switch
+          value={value}
+          onValueChange={() => {
+            // Sends the new checked value
+            return onValueChange(!value);
+          }}
+          trackColor={s.switchTrackColor}
```

Comment:
Is this what you're going for? 
```suggestion
          trackColor={ Platform.OS === 'ios' ? theme.purple : theme.theme === "dark" ? theme.red2 : theme.base4}
```

Reply:
https://reactnative.dev/docs/switch
you can run the example as a Snack here: https://reactnative.dev/docs/switch
I mucked around with the on android and iOS to try to understand how it all worked

Reply:
Thanks for the example. So, the prop is expecting an object. I see how/why what you put in the stylesheet was working for you- it's just not how a stylesheet is intended to be used. You will want to just pass an object:
```
trackColor={{
 true: theme.purple,
 false: theme.theme === "dark" ? theme.red2 : theme.base4, // Not used on iOS
}}
```
Rather than having two props for the possible track position colors, the API is expecting an object. Another case where the intellisense we would get from Typescript would be useful as I agree calling a prop "color" does not make it obvious it is expecting a prop of type object. 

It's not uncommon for libraries to have differences in the component APIs between platforms. Without looking into the library code, I'm uncertain why there is that extra prop.

Reply:
Thanks for the explanation and patience. Stylesheets have always been a pain point for me.